client/components/shipping/packages/add-package.js

```diff
@@ -10,6 +10,11 @@ import AddPackagePresets from './add-package-presets';
 import { translate as __ } from 'lib/mixins/i18n';
 
 const getDialogButtons = ( mode, dismissModal, savePackage, packageData ) => {
+	// replace empty string with null so that reducer removes the field altogether
+	if ( '' === packageData.outer_dimensions ) {
+		delete packageData.outer_dimensions;
+	}
+
```

Comment:
I don't think `getDialogButtons()` is the right place for this logic - how about moving it to the `AddPackageDialog` render method?